### PR TITLE
coredns: v1.7.1

### PR DESCRIPTION
**Pull request by dweomer**
Part of rancher/rke2#345

### CODE REVIEW

File: packages/rke2-coredns/rke2-coredns.patch [hunk 1]
```diff
@@ -167,6 +176,150 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-coredns/charts-original/templates
...
++            {{ end }}
++{{- range .Values.extraSecrets }}
++        - name: {{ .name }}
++          secret:
++            secretName: {{ .name }}
++            defaultMode: 400
++{{- end }}
++{{- if .Values.extraVolumes }}
++{{ toYaml .Values.extraVolumes | indent 8 }}
++{{- end }}
```
Comment (dweomer) [inline packages/rke2-coredns/rke2-coredns.patch:322]: This is the part that worries me: its replacing the deployment.yaml, I think?

File: packages/rke2-coredns/rke2-coredns.patch [hunk 2]
```diff
@@ -314,24 +467,68 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-coredns/charts-original/templates
...
-@@ -3,8 +3,8 @@
+@@ -3,7 +3,7 @@
  # Declare variables to be passed into your templates.
  
  image:
 -  repository: coredns/coredns
--  tag: "1.6.9"
 +  repository: rancher/hardened-coredns
-+  tag: "v1.6.9"
+   tag: "1.7.1"
```
Comment (brandond) [inline packages/rke2-coredns/rke2-coredns.patch:527]: The actual tag here seems to be `v1.7.1`
https://hub.docker.com/layers/rancher/hardened-coredns/v1.7.1/images/sha256-2f943b97ae07055f177f9c89a429bf8a2b0afd85a4f85abc51def84773fef814
```suggestion
   tag: "v1.7.1"
```

**Review by brandond — CHANGES_REQUESTED** [review]
I can't make heads or tails of what this is actually deploying for a corefile, but there is a change necessary between v1.6.9 and v1.7.1, see
https://github.com/k3s-io/k3s/commit/cd27c6fcbe5fe266b20f682374f09cf52661ceeb#diff-1c144ada012fb4c7c809e2fbddfac3fa2acbae57d85ebbce654c94944f667252L61

File: packages/rke2-coredns/rke2-coredns.patch (hunk 2 above)
Comment (dweomer) [reply packages/rke2-coredns/rke2-coredns.patch:527]: Yes, but, https://hub.docker.com/layers/rancher/hardened-coredns/v1.6.9/images/sha256-97f077ce82448672f87a98b54a8da5ebd46ca01aeb4be6e59990399b470a0865?context=explore

File: packages/rke2-coredns/rke2-coredns.patch [hunk 3]
```diff
@@ -314,24 +467,68 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rke2-coredns/charts-original/templates
...
    pullPolicy: IfNotPresent
  
- replicaCount: 1
-@@ -34,10 +34,10 @@
-     prometheus.io/port: "9153"
+@@ -66,10 +66,10 @@
+   annotations: {}
  
  serviceAccount:
```
Comment (briandowns) [inline packages/rke2-coredns/rke2-coredns.patch:533]: Do we want to flip this to true and include the name field like we currently have?

File: packages/rke2-coredns/rke2-coredns.patch (hunk 2 above)
Comment (dweomer) [reply packages/rke2-coredns/rke2-coredns.patch:527]: Oh, I see what i did. :bonk:

File: packages/rke2-coredns/rke2-coredns.patch (hunk 3 above)
Comment (brandond) [reply packages/rke2-coredns/rke2-coredns.patch:533]: I can't tell what this patch is actually doing - are we dropping the replicaCount, and Prometheus scrape port annotations?

Comment (dweomer) [reply packages/rke2-coredns/rke2-coredns.patch:533]: no, its just the change is no longer anchored to that text

Comment (dweomer) [reply packages/rke2-coredns/rke2-coredns.patch:533]: @briandowns create service account is being set to true via the values file

**Pull request merged**